docs/html/topics/vcs-support.md

```diff
@@ -139,9 +139,16 @@ option.
 pip looks at 2 fragments for VCS URLs:
 
 - `egg`: For specifying the "project name" for use in pip's dependency
-  resolution logic. eg: `egg=project_name`
+  resolution logic. e.g.: `egg=project_name`
+
+  The `egg` fragment may additionally contain an extras specifier, e.g.:
+  `egg=project_name[dev,test]`.
+
+  Both the project name and extras specifier must appear in the form
+  defined by [PEP 508](https://peps.python.org/pep-0508/).
```

Comment:
N.B.: I intentionally didn't try to define the semantics/behavior of extras in the egg fragment here, only how they're parsed. But I can add that too, if desired.

Reply:
Happy if you want to leave it at just syntax.